Description: [PATCH] Work around broken 1284 device IDs that use a newline instead of a semicolon.
Date: Tue, 9 Oct 2012 03:01:31 +0000
Origin: http://svn.cups.org/public/cups/trunk@10642
Last-Update: 2012-11-06

--- a/backend/snmp.c
+++ b/backend/snmp.c
@@ -1025,6 +1025,11 @@
 	    * Description is the IEEE-1284 device ID...
 	    */
 
+            char *ptr;			/* Pointer into device ID */
+
+            for (ptr = (char *)packet.object_value.string.bytes; *ptr; ptr ++)
+              if (*ptr == '\n')
+                *ptr = ';';		/* A lot of bad printers put a newline */
 	    if (!device->id)
 	      device->id = strdup((char *)packet.object_value.string.bytes);
 
@@ -1066,8 +1071,11 @@
 	  */
 
 	  char	make_model[256];	/* Make and model */
+          char *ptr;			/* Pointer into device ID */
 
-
+          for (ptr = (char *)packet.object_value.string.bytes; *ptr; ptr ++)
+            if (*ptr == '\n')
+              *ptr = ';';		/* A lot of bad printers put a newline */
 	  if (device->id)
 	    free(device->id);
 
